mailpoet/assets/js/src/automation/editor/index.tsx

```diff
@@ -72,26 +72,45 @@ function updatingActiveWorkflowNotPossible() {
   );
 }
 
+function useConfirmUnsaved(isSaved) {
+  function onUnload(event) {
+    if (!isSaved) {
+      event.returnValue = __('changesNotSaved', 'mailpoet'); // eslint-disable-line no-param-reassign
+      return event.returnValue;
+    }
+    return '';
+  }
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', onUnload);
+    return () => window.removeEventListener('beforeunload', onUnload);
+  });
```

Comment:
I think having no deps array here will result in adding and removing the listener on every re-render. I think there should be a dependency array. But it's interesting the linter doesn't say anything 🤔

It may not be a big deal in this case but I think the correct dependency array for the `useEffect` call is `[onUnload]`. But since the function is redefined on each render, it won't help. It would also need `useCallback` to keep a stable identity:

```ts
const onUnload = useCallback(() => ..., [isSaved]);
```

But thinking further, even that is problematic, as when the `onUnload` changes, the `removeEventListener` will fail to remove it on unmount as it will be a different function. I think here is a very similar issue: https://stackoverflow.com/questions/53750619/event-listener-is-not-removed-in-useeffect-call-within-function-based-component

So I'm thinking... can we extract `onUnload` to the global level, remove its parameter and use `select(...)` in it instead? Haven't tried that approach, just thinking whether something along those lines could work.